crates/rust-analyzer/src/config.rs

```diff
@@ -3203,8 +3340,10 @@ fn schema(fields: &[SchemaField]) -> serde_json::Value {
         .iter()
         .map(|(field, ty, doc, default)| {
             let name = field.replace('_', ".");
-            let category =
-                name.find('.').map(|end| String::from(&name[..end])).unwrap_or("general".into());
+            let category = name
+                .split_once(".")
+                .map(|(category, _name)| to_title_case(category))
+                .unwrap_or("rust-analyzer".into());
```

Comment:
Defaut changed to put all non-grouped settings at the top level instead of under a "General" category. (Casing is important here, it has to match the case of the extension display name.)